communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,219 @@
+            ProjectVersionRef ref) throws GalleyMavenException, URISyntaxException,
+            CartoDataException {
+
+        MavenPomReader pomReader = carto.getGalley().getPomReader();
+        MavenPomView pomView = pomReader.read(ref, getRepoLocations(tempDir));
+
+        Location location = new SimpleLocation("file:" + tempDir.getAbsolutePath());
+        URI src = new URI(location.getUri());
+
+        DiscoveryConfig disConf = new DiscoveryConfig(src);
+        disConf.setIncludeBuildSection(false);
+        disConf.setIncludeManagedDependencies(false);
+        disConf.setIncludeManagedPlugins(false);
+
+        MavenModelProcessor processor = new MavenModelProcessor();
+        return processor.readRelationships(pomView, src, disConf);
+    }
+
+    // TODO: get the list of locations from main pom.xml
+    private List<? extends Location> getRepoLocations(File cacheDir) {
+        Location location = new SimpleLocation("file:" + cacheDir.getAbsolutePath());
+        List<? extends Location> repoLocations = Arrays.asList(location, new SimpleLocation(
+                "central", "http://repo.maven.apache.org/maven2/"), new SimpleLocation("eap",
+                "http://maven.repository.redhat.com/techpreview/all"), new SimpleLocation(
+                "sonatype", "https://oss.sonatype.org/content/repositories"));
+
+        return repoLocations;
+    }
+
+    /**
+     * Given the locations of the pom.xml files, setup the tempDir into a structure
+     * where Cartographer will be able to analyze the poms, and return back a
+     * list of ProjectVersionRefs objects generated from the pom.xml files.
+     * @param tempDir
+     * @param poms
+     * @param carto
+     * @return
+     * @throws TransferException
+     * @throws IOException
+     */
+    private List<ProjectVersionRef> setupRepositoryDirectoryFromClasspath(File tempDir,
+            List<String> poms, CartographerCore carto) throws TransferException, IOException {
+
+        List<ProjectVersionRef> projectVersionRefs = new LinkedList<>();
+
+        for (String pom : poms) {
+            File pomFile = new File(pom);
+            PomPeek peek = new PomPeek(pomFile);
+            projectVersionRefs.add(peek.getKey());
+
+            String path = ArtifactPathUtils.formatArtifactPath(peek.getKey().asPomArtifact(), carto
+                    .getGalley().getTypeMapper());
+
+            File f = new File(tempDir, path);
+            f.getParentFile().mkdirs();
+            FileUtils.copyFile(pomFile, f);
+        }
+        return projectVersionRefs;
+    }
+
+    /**
+     * Return all the pom.xml paths found under the folderDir directory
+     *
+     * @param folderDir directory to analyze dependencies
+     * @return List of the pom.xml paths
+     */
+    private List<String> findAllPomFiles(File folderDir) {
+
+        Collection<File> pomFilePaths = FileUtils.listFiles(folderDir, new String[]{"xml"}, true);
+
+        List<String> paths = pomFilePaths.stream()
```

Comment:
Just a question: Does it make sense to use parallelStream here? As it works with the file system?
@thescouser89 @janinko What do you think?